internal/services/eventhub/eventhub_namespace_customer_managed_key_resource.go

```diff
@@ -143,7 +143,11 @@ func resourceEventHubNamespaceCustomerManagedKeyCreateUpdate(d *pluginsdk.Resour
 
 		isIdentityAssignedToParent := false
 		for item := range namespace.Identity.IdentityIds {
-			if item == userAssignedIdentity {
+			parentEhnUaiId, err := commonids.ParseUserAssignedIdentityIDInsensitively(item)
+			if err != nil {
+				return fmt.Errorf("parsing %q as a User Assigned Identity ID: %+v", item, err)
+			}
+			if parentEhnUaiId.ID() == userAssignedIdentity {
```

Comment:
Rather than comparing these two IDs by checking the results of `.ID()` match - can we compare the Resource ID values instead:

```suggestion
			if resourceids.Match(parentEhnUaiId, userAssignedIdentity) {
```

The `Match` function is available in https://github.com/hashicorp/go-azure-helpers/pull/234 fwiw

Reply:
https://github.com/hashicorp/terraform-provider-azurerm/pull/25873